### PR TITLE
Indirect Corrections Calculate Paalman Pings - Validate chemical formula

### LINKED ISSUE #13970: Indirect Corrections required Chemical formula on Can Details
Currently if you select use can in the CalculatePaalmanPings tab, the Chemical Formula for Can Details does not have a star next to it indicating it is required.

### PR DESCRIPTION
**Pull request by ElliotAOram**
The Chemical Formula input fields for Sample and Container in the Calculate Paalman Pings interface now checks that the Chemical formula is valid when text is edited.
Additional update to use `const auto` where applicable

**To test:**
- Open Calculate Paalman Pings (Interfaces > Indirect > Corrections > Calculate Paalman Pings)
- Check the `Use Can` option
- Using the sample: `IRS26176_graphite002_red` and container: `IRS26174_graphite002_red` (available from `\\olympic\Babylon5\Public\ElliotOram\Corrections\CalculatePaalmanPings\IRIS\correction_set`) make the interface match the image below
- image
- Try to enter a valid chemical for sample and container (the `*` should be removed in each case)
- Remove those or enter an invalid chemical formula (the `*` should return)
- Ensure that the interface runs when all chemical formula are valid
- Ensure a descriptive error message is produced when chemicals are missing/invalid

Fixes #13970 

Changes only affect the way the user sees the validation (update to the red `*`) hence does not need to be in the release notes.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [x] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by KarlPalmen** [comment]
I found example chemical formats at http://docs.mantidproject.org/nightly/algorithms/SetSampleMaterial-v1.html#chemical-composition-with-examples .

**Comment by ElliotAOram** [comment]
@KarlPalmen sorry I forgot to mention it in the PR description in this case the sample is `H2-O` and the container is `V`. I'll update the description with this info

**Comment by KarlPalmen** [comment]
:shipit:

File: MantidQt/CustomInterfaces/src/Indirect/CalculatePaalmanPings.cpp [hunk 1]
```diff
@@ -402,16 +411,16 @@ void CalculatePaalmanPings::addShapeSpecificSampleOptions(IAlgorithm_sptr alg,
...
-    double canBackThickness = m_uiForm.spFlatCanBackThickness->value();
+    const auto canBackThickness = m_uiForm.spFlatCanBackThickness->value();
     alg->setProperty("SampleThickness", canBackThickness);
   } else if (shape == "Cylinder") {
-    double canOuterRadius = m_uiForm.spCylCanOuterRadius->value();
+    const auto canOuterRadius = m_uiForm.spCylCanOuterRadius->value();
     alg->setProperty("CanOuterRadius", canOuterRadius);
   } else if (shape == "Annulus") {
-    double canOuterRadius = m_uiForm.spAnnCanOuterRadius->value();
+    const auto canOuterRadius = m_uiForm.spAnnCanOuterRadius->value();
```
Comment (FedeMPouzols) [inline MantidQt/CustomInterfaces/src/Indirect/CalculatePaalmanPings.cpp]: :+1:

**Pull request merged**